### ISSUE TITLE
Improve plain console logging when executing work in parallel

### ISSUE THREAD

**Issue by big-guy** [gradle/gradle-native#436]

**Comment by lacasseio** [comment]
This PR may help us achieve this work: https://github.com/gradle/gradle/pull/4537

It was part of the discussion during the rich console work for 4.0 but was too big to tackle at the time.

**Comment by adammurdoch** [comment]
In looking through the code with @ghale, we noticed that `OutputEventRenderer` sets up a couple of handlers for plain console listeners. However, if the `OutputEventRenderer` is used in a context where there are no plain console listeners, the handlers still do a bunch of work and then throw the result away. Some examples of this would be in the client when the rich console is used, or in worker processes, which never have a console.

We should fix this if we have time in this issue, or raise as another issue if we don't have time

**Comment by ghale** [comment]
Here's a PR for the first set of work in adding task grouping: https://github.com/gradle/gradle/pull/4713.  This adds a point where we attach a "plain console" which is a throttled output event chain similar to the "default" synchronous output event chain.  It does not change the output at all yet, but the next step would be to implement a `GroupedProgressLogEventGenerator` into this chain to get the task grouping.

**Comment by ghale** [comment]
Here's a WIP PR to actually add the task grouping: https://github.com/gradle/gradle/pull/4756

Here's an example of what the plain console output now looks like: https://asciinema.org/a/CoenXZM2W1oaCXELTWIe9ICw4

**Comment by ghale** [comment]
Un updated example after the latest tweaks: https://asciinema.org/a/Kg7PuAZYXfixaMHdOHhJxTtVp

**Comment by adammurdoch** [comment]
After addressing the review issues and checking the docs, I reckon we should close this issue as done and do whatever follow-on work we need to do as separate issues.

**Comment by ghale** [comment]
I've addressed review items in https://github.com/gradle/gradle/pull/4781

**Comment by ghale** [comment]
The review items PR has been merged.
